core/pbcc_revoke_token.c

```diff
@@ -0,0 +1,148 @@
+    if (uuid) { ADD_URL_PARAM(qparam, uuid, uuid); }
+
+#if PUBNUB_CRYPTO_API
+  ADD_TS_TO_URL_PARAM();
+  SORT_URL_PARAMETERS(qparam);
+#endif
+    ENCODE_URL_PARAMS_TRANS(pt, pb, qparam);
+#if PUBNUB_CRYPTO_API
+    if (pb->secret_key != NULL) {
+        rslt = pbcc_sign_url(pb, token, pubnubUseDELETE, true);
+        if (rslt != PNR_OK) {
+            PUBNUB_LOG_ERROR("pbcc_sign_url failed. pb=%p, message=%s, method=%d", pb, token, pubnubUseDELETE);
+        }
+    }
+#endif
+    PUBNUB_LOG_DEBUG("pbcc_revoke_token_prep. REQUEST =%s\n", pb->http_buf);
+
+    return (rslt != PNR_OK) ? rslt : PNR_STARTED;
+}
+
+
+pubnub_chamebl_t pbcc_get_revoke_token_response(struct pbcc_context* pb)
+{
+    pubnub_chamebl_t result;
+    char const* reply = pb->http_reply;
+    int replylen = pb->http_buf_len;
+
+    if (pb->last_result != PNR_OK) {
+        PUBNUB_LOG_ERROR("pbcc_get_revoke_token(pb=%p) can be called only if "
+                         "previous transactin PBTT_REVOKE_TOKEN(%d) is finished successfully. "
+                         "Transaction result was: %d('%s')\n",
+                         pb,
+                         PBTT_REVOKE_TOKEN,
+                         pb->last_result,
+                         pubnub_res_2_string(pb->last_result));
+        result.ptr = NULL;
+        result.size = 0;
+        return result;
+    }
+
+    result.ptr = (char*)reply;
+    result.size = replylen;
+    return result;
+}
+
+
+enum pubnub_res pbcc_parse_revoke_token_response(struct pbcc_context* pb)
+{
+    char* reply    = pb->http_reply;
+    int   replylen = pb->http_buf_len;
+    struct pbjson_elem elem;
+    struct pbjson_elem parsed;
+    enum pbjson_object_name_parse_result json_rslt;
+
+    if ((replylen < 2) || (reply[0] != '{')) {
+        return PNR_FORMAT_ERROR;
+    }
+    
+    pb->msg_end = replylen + 1;
+
+    elem.end = pbjson_find_end_element(reply, reply + replylen);
+    /* elem.end has to be just behind end curly brace */
+    if ((*reply != '{') || (*(elem.end++) != '}')) {
+        PUBNUB_LOG_ERROR("pbcc_parse_revoke_token_response(pbcc=%p) - Invalid: "
+                         "response from server is not JSON - response='%s'\n",
+                         pb,
+                         reply);
+
+        return PNR_FORMAT_ERROR;
+    }
+    elem.start = reply;
+
+    if (pbjson_value_for_field_found(&elem, "status", "403")){
+        PUBNUB_LOG_ERROR("pbcc_parse_revoke_token_api_response(pbcc=%p) - AccessDenied: "
+                         "response from server - response='%s'\n",
+                         pb,
+                         reply);
+        return PNR_ACCESS_DENIED;
+    }
+
+    json_rslt = pbjson_get_object_value(&elem, "data", &parsed);
+    if (jonmpKeyNotFound == json_rslt) {
+        json_rslt = pbjson_get_object_value(&elem, "error", &parsed);
+        if (json_rslt != jonmpOK) {
+            PUBNUB_LOG_ERROR("pbcc_parse_revoke_token_api_response(pbcc=%p) - Invalid response: "
```

Comment:
@650elx I believe here should be `pbcc_parse_revoke_token_response` instead of `pbcc_parse_revoke_token_api_response` or actual function name should be changed.

Reply:
👌